query-engine/connectors/query-connector/src/filter/scalar.rs

```diff
@@ -65,6 +65,10 @@ impl ScalarFilter {
         self.len() > chunk_size
     }
 
+    pub fn can_batch(&self) -> bool {
+        !matches!(self.condition, ScalarCondition::NotIn(_))
```

Comment:
I believe this is only half the story. While yes, the ticket specifies only `notIn` to be problematic, my assumption right now is that any negation filter will cause problems with splits.